src/lib.rs

```diff
@@ -1142,4 +1213,75 @@ mod tests {
         assert_eq!(di.size_hint(), (2, Some(2)));
         assert_eq!(di.count(), 2);
     }
+
+    #[test]
```

Comment:
Loving these tests — thanks for taking the time!

Reply:
Happy you like them!